frontend/scripts/check-ds-spacing-roles.test.sh

```
@@ -0,0 +1,311 @@
+.host-surface .x-guest {
+  margin-top: var(--gapCards);
+}
+CSS
+  printf '.screen-root {\n  display: grid;\n}\n' >"$core/screens/x.css"
+}
+
+# Runs the gate over a fixture core tree whose screen stylesheet carries $2,
+# leaving the output in $TMP/out. Each case gets its own tree, so one case's
+# rules cannot be read as another's.
+run_case() {
+  local name="$1" rules="$2" core="$TMP/$name"
```

Shellcheck (0.11.0)

[warning] This assignment is used again in this 'local', but won't have taken effect. Use two 'local's.

(SC2318)